src/main/java/com/minecolonies/core/entity/ai/workers/crafting/AbstractEntityAICrafting.java

```diff
@@ -556,7 +562,18 @@ protected IAIState craft()
                     return START_WORKING;
                 }
                 recordCraftingBuildingStats(currentRequest, currentRecipeStorage);
+
                 currentRequest.addDelivery(currentRecipeStorage.getPrimaryOutput());
+
+                // If the primary output is empty, it is not returned as part of the result list
+                // Hence, if the primary output is empty, we can start iterating from index 0
+                // If it's not empty, it is added to the beginning of the stacks list, so we start iterating from index 1
+                for (int i = ItemStackUtils.isEmpty(currentRecipeStorage.getPrimaryOutput()) ? 0 : 1; i < addedStacks.size(); i++)
```

Comment:
Why can't we just do currentRecipestorage.getSecondaryOutputs?

Reply:
That doesn't include the items generated by loot tables.

Reply:
It was slightly finnicky, it were easier if I wouldn't have to again filter out the primary output, but this was the easiest way I could tell to do it

Reply:
We can't, it's always 1 item, and if it were to change we'd have to refactor much more

Reply:
Can we instead just do a for each loop and if does not match primary output continue? I feel that would be cleaner

Reply:
I figured that too, but that has the chance to catch false positives.

Let's say you had a crafting recipe that is guaranteed to make item A once, but like a 50% dice roll to produce the same item again.

That'd net you 2x an itemstack of A with 1 item, if I were to filter out all the items that match the primary output. It'd filter out potential extra drops from the secondary outputs (or loot table) too, which is not what it's supposed to do.

I only need to extract the first item (if it's there)

Reply:
I feel this is a primary output then as well

Reply:
I mean not really, that is chance based, which the primary output cannot suffice. You can only set a hard "this recipe produces X", that part is then tracked and sent as part of the delivery.

Any extras that it produces, via loot table or secondaries are not part of the initial delivery, even if it is the same item as the primary item stack.